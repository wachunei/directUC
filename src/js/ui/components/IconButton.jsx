import styled from "styled-components";

const IconButton = styled.button`
  all: unset;
  cursor: pointer;
  width: 15px;
  height: 15px;
  transition: filter 0.2s;

  &:hover {
    filter: brightness(50%);
  }

  &:focus:not(:hover) {
    outline-width: 2px;
  }

  &:focus:hover {
    outline: none;
  }
`;

export default IconButton;
