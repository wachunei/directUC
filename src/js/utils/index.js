export const removeSuffix = (suffix, value) =>
  value.endsWith(suffix) ? value.slice(0, -suffix.length) : value;

export const parseDistinguishedName = (name) =>
  name.replace(
    /\w\S*/g,
    (txt) => txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase()
  );

export const isFirefox = !window.chrome.app;
