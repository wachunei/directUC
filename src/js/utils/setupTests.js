// eslint-disable-next-line import/no-extraneous-dependencies
import "jest-styled-components";
