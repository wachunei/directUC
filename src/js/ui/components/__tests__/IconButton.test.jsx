import React from "react";
import IconButton from "../IconButton";
import { renderTree } from "./utils";
import "jest-styled-components";

describe("IconButton Component", () => {
  test("should match snapshot", () => {
    const tree = renderTree(<IconButton />);
    expect(tree).toMatchSnapshot();
  });
});
