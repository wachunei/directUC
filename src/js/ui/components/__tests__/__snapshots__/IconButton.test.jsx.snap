// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton Component should match snapshot 1`] = `
.c0 {
  all: unset;
  cursor: pointer;
  width: 15px;
  height: 15px;
  -webkit-transition: filter 0.2s;
  transition: filter 0.2s;
}

.c0:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

.c0:focus:not(:hover) {
  outline-width: 2px;
}

.c0:focus:hover {
  outline: none;
}

<button
  className="c0"
/>
`;
